src/parse.jl

```diff
@@ -0,0 +1,254 @@
+            else
+                error("node $(nodename(field)) unknown")
+            end
+        end
+    end
+    return HDict
+end
+
+"""
+    add_variable!(variables, field, id=1)
+
+### Input
+
+- `variables` -- vector of symbolic variables
+- `field`     -- node with a `param` variable field
+- `id`        -- (optional, default: 1) integer that identifies the variable
+
+### Output
+
+The updated vector of symbolic variables.
+
+### Notes
+
+Parameters can be either variable names (type "real") or labels (type "label").
+"""
+function add_variable!(variables, field, id=1)
+    @assert field["type"] == "real"
+
+    # d1 and d2 are the dimensions (d1=d2=1 for scalars)
+    @assert field["d1"] == "1" && field["d2"] == "1"
+
+    varname = parse(field["name"])
+    islocal = haskey(field, "local") ? parse(field["local"]) : false
+    iscontrolled = haskey(field, "controlled") ? parse(field["controlled"]) : true
+    dynamicstype = haskey(field, "dynamics") ? field["dynamics"] : "any"
+
+    variables[varname] = Dict("local"=>islocal,
+                              "controlled"=>iscontrolled,
+                              "dynamics"=>dynamicstype,
+                              "id"=>id)
+    return variables
+end
+
+"""
+    add_transition_label!(labels, field)
+
+### Input
+
+- `labels` -- vector of transition labels
+- `field`  -- node with a `param` label field
+
+### Output
+
+The updated vector of transition labels.
+
+### Notes
+
+Parameters can be either variable names (type "real") or labels (type "label").
+"""
+function add_transition_label!(labels, field)
+    @assert field["type"] == "label"
+    push!(labels, field["name"])
+    return labels
+end
+
+"""
+    parse_location(field)
+
+### Input
+
+- `field`  -- location node
+
+### Output
+
+The tuple `(id, I, f)` where `id` is the integer that identifies the location,
+`I` is the list of invariants for this location, and `f` is the list of ODEs that
+define the flow. Both objects are vectors of symbolic expressions `Expr`.
+"""
+function parse_location(field)
+    id = parse(Int, field["id"])
+    local I, f
+    for element in eachelement(field)
+        if nodename(element) == "invariant"
+            I = parse_sxmath(nodecontent(element))
+        elseif nodename(element) == "flow"
+            f = parse_sxmath(nodecontent(element))
+        else
+            warn("field $(nodename(element)) in location $(field["id"]) is ignored")
+        end
+    end
+    return (id, I, f)
+end
+
+"""
+    parse_transition(field)
+
+### Input
+
+- `field`  -- transition node
+
+### Output
+
+The tuple `(q, r, G, A)` where `q` and `r` are the source mode and target mode
+respectively for this location, `G` is the list of guards for this location,
+and `A` is the list of assignments. Both objects are vectors of
+symbolic expressions `Expr`.
+
+### Notes
+
+It is assumed that the "source" and "target" fields can be cast to integers.
```

Comment:
This sounds strict. Is this reasonable?
Otherwise you could store the transitions in a temporary data structure and obtain the location IDs later from a dictionary "location name → location ID".

Reply:
This is given by the format, in that id's are assigned automatically (for locations, transitions, etc). I don't remember if there is a way of editing these fields in the model editor.